File: 백준/Silver/14716. 현수막/현수막.kt
package org.example

private val directions = arrayOf(
    intArrayOf(-1, -1),
    intArrayOf(-1, 0),
    intArrayOf(-1, 1),
    intArrayOf(0, -1),
    intArrayOf(0, 1),
    intArrayOf(1, -1),
    intArrayOf(1, 0),
    intArrayOf(1, 1),
)

fun dfs(graph: Array<IntArray>, x: Int, y: Int) {
    graph[x][y] = 2 // 방문 표시
    for ((dx, dy) in directions) {
        val nx = x + dx
        val ny = y + dy
        if (nx in 0 until graph.size && ny in 0 until graph[0].size && graph[nx][ny] == 1) {
            dfs(graph, nx, ny)
        }
    }
}

fun main() {
    val br = System.`in`.bufferedReader()
    val (m, n) = br.readLine().split(" ").map { it.toInt() }
    val graph = Array(m) {
        br.readLine().split(" ").map { it.toInt() }.toIntArray()
    }

    var count = 0
    for (i in 0 until m) {
        for (j in 0 until n) {
            if (graph[i][j] == 1) {
                dfs(graph, i, j)
                count++
            }
        }
    }
    println(count)
    br.close()
}

File: 백준/Silver/14716. 현수막/현수막.py
import sys
from collections import deque

input = sys.stdin.readline

directions = [
    (-1, -1), (-1, 0), (-1, 1),
    (0, -1), (0, 1),
    (1, -1), (1, 0), (1, 1)
]


def bfs(graph, x, y):
    queue = deque([(x, y)])
    graph[x][y] = 2  # 방문 표시
    while queue:
        cx, cy = queue.popleft()
        for (dx, dy) in directions:
            nx = cx + dx
            ny = cy + dy
            if 0 <= nx < len(graph) and 0 <= ny < len(graph[0]) and graph[nx][ny] == 1:
                queue.append((nx, ny))
                graph[nx][ny] = 2


# 행 m, 열 n
m, n = map(int, input().split())
graph = []
for _ in range(m):
    graph.append(list(map(int, input().split())))

count = 0
for i in range(m):
    for j in range(n):
        if graph[i][j] == 1:
            bfs(graph, i, j)
            count += 1

print(count)
